### PR TITLE
fix: add type checking to templates

### PR DESCRIPTION
**Pull request by zoltantakacs-born**
Type checking didn't work for some of the Vue templates.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by scernisevs — APPROVED** [review]
LGTM

**Review by dhargitai — APPROVED** [review]
Looks good

File: packages/composables/src/types/useCategory/index.ts [hunk 1]
```diff
@@ -1,64 +1,3 @@
-import {
```
Comment (WojtekTheWebDev) [inline packages/composables/src/types/useCategory/index.ts:1]: We're not using those types?

File: packages/theme/components/LoginModal.vue [hunk 2]
```diff
@@ -216,7 +223,16 @@ export default {
     const SCREEN_REGISTER = 'register';
 
     const { isLoginModalOpen, toggleLoginModal } = useUiState();
-    const form = ref({});
+    const form = ref(
+      {} as {
```
Comment (WojtekTheWebDev) [inline packages/theme/components/LoginModal.vue:227]: You're using TS here, so you can use ref<{IForgotWhatTypeItWasButForSureItCanBeImprotedFromUserTypes}>();

**Review by WojtekTheWebDev — CHANGES_REQUESTED** [review]
Nice job, just one suggestion with usage of `ref` and one question

File: packages/composables/src/types/useCategory/index.ts (hunk 1 above)
Comment (zoltantakacs-born) [reply packages/composables/src/types/useCategory/index.ts:1]: No. We use the Category and CategoryTree types coming from the bigcommerce-api package.

**Review by WojtekTheWebDev — APPROVED** [review]
LGTM

**Review by WojtekTheWebDev — COMMENTED** [review]
LGTM

**Pull request merged**